packages/fast-components-react-base/src/viewport-positioner/viewport-positioner.tsx

```diff
@@ -879,6 +903,44 @@ class ViewportPositioner extends Foundation<
         ) as ViewportPositionerState);
     };
 
+    /**
+     * Get width of positioner for next render
+     */
+    private getNextPositionerWidth = (
+        desiredHorizontalPosition: ViewportPositionerHorizontalPositionLabel
+    ): number => {
+        let newPositionerRectWidth: number = this.positionerRect.width;
+        if (this.props.scaleToFit) {
+            const availableWidth: number = this.getAvailableWidth(
+                desiredHorizontalPosition
+            );
+            newPositionerRectWidth =
+                availableWidth > this.viewportRect.width
+                    ? this.viewportRect.width
+                    : availableWidth;
+        }
+        return newPositionerRectWidth;
+    };
+
+    /**
+     * Get height of positioner for next render
+     */
+    private getNextPositionerHeight = (
+        desiredVerticalPosition: ViewportPositionerVerticalPositionLabel
+    ): number => {
+        let newPositionerRectHeight: number = this.positionerRect.height;
+        if (this.props.scaleToFit) {
+            const availableHeight: number = this.getAvailableHeight(
+                desiredVerticalPosition
+            );
+            newPositionerRectHeight =
+                availableHeight > this.viewportRect.height
+                    ? this.viewportRect.height
+                    : availableHeight;
+        }
+        return newPositionerRectHeight;
+    };
```

Comment:
These functions do the same thing, we can create some abstraction here so that you could combine and do something like this:

```typescript
export enum Dimension {
    height,
    width
}

// inside class

private getNextPositionDimension = (
    dimension: Dimension,
    desiredDimensionPosition: ViewportPositionerPositionLabel // would need to make this generic
): number => {
    if (this.props.scaleToFit) {
        const availableSize: number = this.getAvailableDimension( // would need to make this generic
            desiredDimensionPosition,
            dimension
        );
        return availableSize > this.viewportRect[dimension]
            ? this.viewportRect[dimension]
            : availableSize;
    }

    return this.positionRect[dimension];
}

Reply:
I would like to do a refinement pass with the specific intention of collapsing the axis calculations into more generic functions on this whole class.  I think that should be it's own chore and PR as it could involve a lot of refactoring.

Reply:
We shouldn't really need to plan abstraction in the future if we can just do it now. I think it's worthwhile to make sure the cleanest commits and code goes in the first time so that we don't have to come back to it.